### PR TITLE
Issue #677 Transition CI build to use Pixi

### PR DESCRIPTION
**Pull request by Manangka**
This commit adds the Pixi build system to the project. It enables us to simplify the CI pipeline, speed up the pipeline, debug the pipeline steps locally and install a developer environment with more ease.

The imod-environment.yml and imod-environment-dev.yml files are no longer required and have been removed

### CODE REVIEW

File: pixi.toml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+hypothesis = "*"
+isort = "*"
+jinja2 = "*"
+matplotlib = "*"
+mypy = "*"
+netcdf4 = "*"
+numba = ">=0.50"
+numpy = "*"
+pandamesh = "*"
+pandas = "<2.0"  # TODO: unpin when newer xarray on conda-forge
```
Comment (JoerivanEngelen) [inline pixi.toml:62]: Note to ourselves: We also need to update this file when https://github.com/Deltares/imod-python/pull/715 will be merged

File: pixi.toml [hunk 2]
```diff
@@ -0,0 +1,94 @@
+[project]
+name = "imod-python"
+version = "0.15.1"
+description = "Make massive MODFLOW models!"
```
Comment (JoerivanEngelen) [inline pixi.toml:4]: Minor nitpick: This is already a bold statement, the ``!`` is a bit too much ;-)

```suggestion
description = "Make massive MODFLOW models"
```

File: pixi.toml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+authors = ["Deltares <huite.bootsma@deltares.nl>", ]
```
Comment (JoerivanEngelen) [inline pixi.toml:5]: Something to pick up in a later issue: what to do with this mail address in our aim to "professionalize" the software suite: Add a complete list of authors with full names, like [xmipy](https://github.com/Deltares/xmipy/blob/develop/pyproject.toml) & [hydromt](https://github.com/Deltares/hydromt/blob/main/pyproject.toml)? Or use imod.info@deltares.nl or something, like [ribasimpy](https://github.com/Deltares/Ribasim/blob/main/python/ribasim/pyproject.toml)?

**Review by JoerivanEngelen — APPROVED** [review]
Some comments, but they are more remarks/notes to ourselves than things that need to be fixed in this PR

File: pixi.toml (hunk 2 above)
Comment (Manangka) [reply pixi.toml:4]: Done

**Pull request merged**